src/Chapter4.hs

```diff
@@ -674,6 +711,79 @@ Specifically,
 -}
 
 
+data Btree a = Btree
+  {
+    lchild :: Btree a,
+    rchild :: Btree a,
+    value :: a
+  } | Null deriving (Show)
```

Comment:
That is right representation of the Tree.

However, having record fields inside the sum type considered antipattern and highly discouraged.
As we discussed in previous Chapter, each record name is the function in reality, but as it is the sum type, these functions are partial (can fail in runtime). So it is better to not use that.

In this case you have two options. First, and most common, to use unnamed fields:
```haskell
data Btree a 
    = Btree (Btree a) (BTree a) a 
    | Null 
    deriving (Show)
```

Another option, to use additional record type, and use it unnamed inside the tree

```haskell
data Node a = Node
    { lchild :: Btree a
    , rchild :: Btree a
    , value :: a
    }

data Btree a 
    = Tree (Node a) 
    | Null 
    deriving (Show)
```

Reply:
This is very helpful, thanks!